### DISCUSSION TITLE
new json object add carriage and return

### DISCUSSION THREAD

**Discussion by AndresDuque** [nlohmann/json#3769] [Q&A]
Hi there,

I am working on an API call using the JSON library. While I am creating the JSON object, the carriage and return (/r/n) are being added to it. This makes an error to the API call that I need to perform detecting that some characters are not desired. I already tried it with postman as well and I can't make the API call as well because of this carriage and return.

"data": "{\r\n \"Content-Type\": \"image/png\",\r\n \"acl\": \"private\",\r\n .." That's an example. I should send the data. It's there any way to avoid this? In python, I don't have this issue for example.

Best regards,
Andres.

**Comment by falbrechtskirchinger** [comment]
I can only speculate because you've not given enough information. (Show some code!)

Are you serializing the JSON object to a stream? Looks like you're on Windows and the stream is not set to binary. Also, it appears you're calling `dump()` with a non-zero indentation value (otherwise, the serializer won't output newlines).

**Reply by AndresDuque** [reply]
Hi there! Thanks for the answer. I am receiving a response from AWS and then using the JSON lib to parse it. Then, use the object to call the data I need to send back. This is the code I use for parsing.

json object = json::parse(response.text);
// Afterwards I call the data to send
object["url"], object["data"].. // here is where the back slashes, carriage and return are added.

And then I use the CPR library to send the information back. I opened the discussion because I was receiving that the JSON was containing malformed characters and I just saw that it may be down to another issue (I saw that the JSON was valid even containing back slashes).

**Reply by falbrechtskirchinger** [reply]
That's still not enough information.

```cpp
object["url"], object["data"].. // here is where the back slashes, carriage and return are added.
```

What exactly are you doing here? [`operator []`](https://json.nlohmann.me/api/basic_json/operator%5B%5D/) does not magically add `\r\n` to anything.
What are you passing to CPR? A string? If so, how are you converting your JSON object to a string?
What's the exact error message and are you sure `\r\n` is the real issue in the first place?

**Reply by AndresDuque** [reply]
Well... the JSON is basically a dictionary, so with the operator[] I am just getting the value related to that key. Nothing else. I am passing a string to CPR using dump().

As I said, I was getting that my call was malformed but now I am getting another error not related to JSON. I validated that the JSON object was right. Now, it's another thing. I am going to close the question.
